wal/wal.go

```diff
@@ -480,6 +480,106 @@ func (w *WAL) ReadAll() (metadata []byte, state raftpb.HardState, ents []raftpb.
 	return metadata, state, ents, err
 }
 
+// Verify reads through the records of the current WAL and verifies that it is not corrupt.
+// Unlike ReadAll, it does not return the contents of the WAL.
+// Verify cannot be called twice on the same WAL, or after a ReadAll call.
+// If opened in write mode, it must read out all records until EOF. Or an error
+// will be returned.
+// If opened in read mode, it will try to read all records if possible.
+// It is recommended to call Verify only when WAL is opened in read mode.
+// If it cannot read out the expected snap, it will return ErrSnapshotNotFound.
+// If loaded snap doesn't match with the expected one, it will return
+// all the records and error ErrSnapshotMismatch.
+// After Verify, the WAL will be ready for appending new records.
+func (w *WAL) Verify() error {
```

Comment:
The build failure seems outside the changes made here. But meanwhile we wait for @xiang90 's comment, good idea if you can see that a new test can be added in the wal_test for this new func. Thanks!

Reply:
Sure! I'll write a test for the `Verify` function and push it shortly.